### ISSUE TITLE
Add recovery test using a code id that differs from that in the table of the defunct ledger

### ISSUE THREAD

**Issue by achamayou** [microsoft/CCF#2792]
A user recovering a ledger must be able to use new code.

**Comment by jumaffre** [comment]
I believe this is what the `run_ledger_compatibility_since_first` part of the LTS compatibility test already does.

**Comment by achamayou** [comment]
@jumaffre good point, but I think it would still be good to have a generic, suite-able version (something that did a loggingcpp -> loggingjs recovery, or the other way around as the case may be). The LTS could re-use that with slightly different args?

**Comment by jumaffre** [comment]
@achamayou Yes, that would be even better!

**Comment by achamayou** [comment]
Closed by #7986
